substratest/assets.py

```diff
@@ -67,14 +67,15 @@ def __init__(self, compute_plan, session):
 
     def wait(self, timeout=FUTURE_TIMEOUT):
         """wait until all tuples are completed (done or failed)."""
-        for key in self._compute_plan.traintuple_keys:
-            self._session.get_traintuple(key).future().wait(timeout, raises=False)
-        for key in self._compute_plan.composite_traintuple_keys:
-            self._session.get_composite_traintuple(key).future().wait(timeout, raises=False)
-        for key in self._compute_plan.aggregatetuple_keys:
-            self._session.get_aggregatetuple(key).future().wait(timeout, raises=False)
-        for key in self._compute_plan.testtuple_keys:
-            self._session.get_testtuple(key).future().wait(timeout, raises=False)
+        tuples = (self._compute_plan.list_traintuple(self._session) +
+                  self._compute_plan.list_composite_traintuple(self._session) +
+                  self._compute_plan.list_aggregatetuple(self._session))
+        tuples = sorted(tuples, key=lambda t: t.rank)
+        # testtuples do not have a rank attribute
+        tuples += self._compute_plan.list_testtuple(self._session)
+
+        for tuple in tuples:
+            tuple.future().wait(timeout, raises=False)
```

Comment:
We should avoid using the reserved keyword tuple here. Could be replaced with `tuple_`?

Reply:
sure, my bad.